src/Bicep.Core/Syntax/ImportSpecification.cs

```diff
@@ -46,35 +49,57 @@ private ImportSpecification(string name, string version, bool isValid, TextSpan
 
         public static ImportSpecification From(SyntaxBase specificationSyntax)
         {
-            switch (specificationSyntax)
+            return specificationSyntax switch
             {
-                case StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value:
-                    var (name, version, isValid) = Parse(value);
-                    var span = isValid ? new TextSpan(stringSyntax.Span.Position + 1, name.Length) : stringSyntax.Span;
-
-                    return new ImportSpecification(name, version, isValid, span);
-
-                case SkippedTriviaSyntax trivia:
-                    return new ImportSpecification(trivia.TriviaName, trivia.TriviaName, false, trivia.Span);
+                StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value
+                    => CreateFromStringSyntax(stringSyntax, value),
+                SkippedTriviaSyntax trivia
+                    => new ImportSpecification(
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        false,
+                        trivia.Span),
+                _
+                    => new ImportSpecification(
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        false,
+                        specificationSyntax.Span)
+            };
+        }
 
-                default:
-                    return new ImportSpecification(LanguageConstants.ErrorName, LanguageConstants.ErrorName, false, specificationSyntax.Span);
+        public SyntaxBase ToPath()
+        {
+            if (!this.IsValid)
+            {
+                return new SkippedTriviaSyntax(this.Span, Enumerable.Empty<SyntaxBase>());
             }
+            return SyntaxFactory.CreateStringLiteral(this.UnqualifiedAddress);
```

Comment:
Shouldn't this be fully-qualified?

Reply:
No, the semantic processing logic is going to expand the address later. 
For example:
![image](https://github.com/Azure/bicep/assets/9611108/43d6c2ad-4e82-4588-9655-e80d1f464967)

In this case you can see that the value is `br:public:az:2.0.0`, this is not fully expanded... During the semantic processing in 
https://github.com/Azure/bicep/blob/asilverman/dynamic-type-loading/aliases/src/Bicep.Core/Registry/ModuleDispatcher.cs#L48-L90 

The value is expanded into its fully qualified form. Perhaps a better name is `AddressTrivia`?

Reply:
Yeah I think the naming is confusing me. Correct me if I'm misunderstanding, but from what I can tell there are two concepts of full/un-qualified relating to imports:
* Address:
     * Unqualified: The address can potentially include an alias (e.g. `br/public:...`)
     * Fully-qualified: The alias must have been resolved and point to a fqdn for a registry (e.g. `mcr.microsoft.com...`
* Reference:
    * Fully-qualified: The reference contains a scheme (`ts:...` or `br:...`)
    * Unqualified: The reference does not contain a scheme (`mcr.microsoft.com...`)

Some follow-up questions:
* Is `unqualifiedAddress` also used built-in providers (e.g. `sys:1.0.0`), or is it purely reserved for "things that can go in a registry"?
* Does a "reference" (fully or un-qualified) always contain a fully-qualified address, or can it also contain an unqualified address?

Reply:
Can we discuss offline in a meeting? I feel like the async nature of request/response patterns is making the PR take a long time to resolve and its impacting the ability to implement the feature